x/gamm/types/msgs.go

```diff
@@ -60,6 +64,82 @@ func ValidateFutureGovernor(governor string) error {
 	return nil
 }
 
+var _ sdk.Msg = &MsgSwapExactAmountIn{}
+
+func (msg MsgSwapExactAmountIn) Route() string { return RouterKey }
+func (msg MsgSwapExactAmountIn) Type() string  { return TypeMsgSwapExactAmountIn }
+func (msg MsgSwapExactAmountIn) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Sender)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid sender address (%s)", err)
+	}
+
+	err = swaproutertypes.SwapAmountInRoutes(msg.Routes).Validate()
+	if err != nil {
+		return err
+	}
+
+	if !msg.TokenIn.IsValid() || !msg.TokenIn.IsPositive() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidCoins, msg.TokenIn.String())
+	}
```

Comment:
Do we continue to use sdkerrors here? Just remember you saying something about avoiding their use.

Reply:
I'm not adding new code, only bringing back what was removed. For ease of merging to main, we should keep things as they used to be